File: applicatie/src/main/java/com/vdzon/administratie/overzicht/OverzichtService.java
package com.vdzon.administratie.overzicht;

import com.vdzon.administratie.auth.SessionHelper;
import com.vdzon.administratie.checkandfix.rest.CheckAndFixService;
import com.vdzon.administratie.crud.UserCrud;
import com.vdzon.administratie.model.Gebruiker;
import com.vdzon.administratie.pdfgenerator.GenerateOverzicht;
import com.vdzon.administratie.util.SingleAnswer;
import spark.Request;
import spark.Response;

import javax.inject.Inject;
import java.io.BufferedOutputStream;

public class OverzichtService {

    @Inject
    UserCrud crudService;

    @Inject
    CheckAndFixService checkAndFixService;

    protected Object getPdf(Request req, Response res) throws Exception {
        Gebruiker gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService);
            String beginDate = req.params(":beginDate");
            String endDate = req.params(":endDate");
            if ("endDate".equals(endDate)) {
                res.status(404);
                return new SingleAnswer("not found");
            }
            if ("beginDate".equals(beginDate)) {
                res.status(404);
                return new SingleAnswer("not found");
            }

            res.raw().setContentType("application/pdf");
            res.raw().setHeader("Content-Disposition","attachment; filename=overzicht_"+beginDate+"_"+endDate+".pdf");
            try (BufferedOutputStream zipOutputStream = new BufferedOutputStream(res.raw().getOutputStream())) {
                GenerateOverzicht.buildPdf(gebruiker.getDefaultAdministratie(), beginDate, endDate, zipOutputStream, checkAndFixService);
                zipOutputStream.flush();
                zipOutputStream.close();
            }

            return null;
    }

}

File: rest/src/main/java/com/vdzon/administratie/rest/overzicht/OverzichtService.kt
package com.vdzon.administratie.rest.overzicht

import com.vdzon.administratie.crud.UserCrud
import com.vdzon.administratie.pdfgenerator.factuur.GenerateOverzicht
import com.vdzon.administratie.util.SessionHelper
import com.vdzon.administratie.util.SingleAnswer
import spark.Request
import spark.Response
import java.io.BufferedOutputStream
import javax.inject.Inject

class OverzichtService
@Inject
constructor(val generateOverzicht: GenerateOverzicht, var crudService: UserCrud) {

    @Throws(Exception::class)
    fun getPdf(req: Request, res: Response): Any? {
        val gebruiker = SessionHelper.getGebruikerOrThowForbiddenExceptin(req, crudService)
        val beginDate = req.params(":beginDate")
        val endDate = req.params(":endDate")
        if ("endDate" == endDate) {
            res.status(404)
            return SingleAnswer("not found")
        }
        if ("beginDate" == beginDate) {
            res.status(404)
            return SingleAnswer("not found")
        }

        res.raw().contentType = "application/pdf"
        res.raw().setHeader("Content-Disposition", "attachment; filename=overzicht_" + beginDate + "_" + endDate + ".pdf")
        BufferedOutputStream(res.raw().outputStream).use { zipOutputStream ->
            generateOverzicht.buildPdf(gebruiker.defaultAdministratie, beginDate, endDate, zipOutputStream)
            zipOutputStream.flush()
            zipOutputStream.close()
        }

        return null
    }

}
